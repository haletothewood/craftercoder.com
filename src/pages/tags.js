import React from "react"

import kebabCase from "lodash/kebabCase"

import { Helmet } from "react-helmet"
import { Link, graphql } from "gatsby"

import Layout from "../components/layout"

import "../styles/tag-listing.scss"

export const PureTags = ({
  data: {
    allMarkdownRemark: { group },
    site: {
      siteMetadata: { title },
    },
  },
}) => {
  return (
    <>
      <Helmet title={title} />
      <div>
        <h1>Tags</h1>
        <ul>
          {group.map(tag => (
            <li key={tag.fieldValue}>
              <Link
                className="tag-listing"
                to={`/tags/${kebabCase(tag.fieldValue)}/`}
              >
                {tag.fieldValue} ({tag.totalCount})
              </Link>
            </li>
          ))}
        </ul>
      </div>
    </>
  )
}

export const Tags = props => {
  return (
    <Layout>
      <PureTags {...props} />
    </Layout>
  )
}

export default Tags

export const pageQuery = graphql`
  query {
    site {
      siteMetadata {
        title
      }
    }
    allMarkdownRemark(limit: 2000) {
      group(field: frontmatter___tags) {
        fieldValue
        totalCount
      }
    }
  }
`
