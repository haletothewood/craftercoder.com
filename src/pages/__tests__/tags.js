import React from "react"
import renderer from "react-test-renderer"
import { PureTags as Tags } from "../tags"

const props = {
  data: {
    allMarkdownRemark: {
      group: [
        {
          fieldValue: "Test Tag",
          totalCount: 1,
        },
        {
          fieldValue: "Another Test Tag",
          totalCount: 1,
        },
      ],
    },
    site: {
      siteMetadata: {
        title: "Crafter Coder",
      },
    },
  },
}

describe("Tags", () => {
  it("renders correctly", () => {
    const tree = renderer.create(<Tags {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
