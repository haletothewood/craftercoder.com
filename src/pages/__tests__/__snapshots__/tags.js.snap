// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tags renders correctly 1`] = `
<div>
  <h1>
    Tags
  </h1>
  <ul>
    <li>
      <a
        className="tag-listing"
        href="/tags/test-tag/"
      >
        Test Tag
         (
        1
        )
      </a>
    </li>
    <li>
      <a
        className="tag-listing"
        href="/tags/another-test-tag/"
      >
        Another Test Tag
         (
        1
        )
      </a>
    </li>
  </ul>
</div>
`;
